app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+  validates :title, :text, presence: true
+
+  def update_user_posts_counter
+    author.posts_counter = author.posts.count
+    author.save
+  end
```

Comment:
- [ ] [optional] @Salwa99 great work on your update  counter method, however, I would recommend you t use  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_user_posts_counter method every time a new post has been saved to the database, then make the update_user_posts_counter method private